tmt/steps/discover/__init__.py

```diff
@@ -146,55 +151,70 @@ def save(self) -> None:
         super().save()
 
         # Create tests.yaml with the full test data
-        tests = dict([
-            (test.name, test.export(format_='dict'))
-            for test in self.tests()])
-        self.write('tests.yaml', tmt.utils.dict_to_yaml(tests))
+        raw_test_data = {
+            test.name: test.export(format_='dict')
+            for test in self.tests()
+            }
+
+        self.write('tests.yaml', tmt.utils.dict_to_yaml(raw_test_data))
 
         # Create 'run.yaml' with the list of tests for the executor
         if not self.tests():
             return
-        tests = dict([
-            (test.name, test.export(format_='execute'))
-            for test in self.tests()])
-        self.write('run.yaml', tmt.utils.dict_to_yaml(tests, width=1000000))
+        raw_test_data = {
+            test.name: test.export(format_='execute')
+            for test in self.tests()
+            }
+        self.write('run.yaml', tmt.utils.dict_to_yaml(raw_test_data, width=1000000))
```

Comment:
Interesting. I'd rather keep it and remove it later, to avoid "hiding" the change into this particular patch set. It already contains so many changes :/

Reply:
Understood, created a separate pull request #1511 with this change.